src/algorithm/boundingbox.rs

```diff
@@ -0,0 +1,209 @@
+use num::{Float};
+
+use types::{Bbox, Point, MultiPoint, LineString, MultiLineString, Polygon, MultiPolygon};
+
+/// Calculation of the bounding box of a geometry.
+
+pub trait BoundingBox<T: Float> {
+    /// Return a Bounding Box of a geometry
+    ///
+    /// ```
+    /// use geo::{Point, LineString, Coordinate};
+    /// use geo::algorithm::boundingbox::BoundingBox;
+    ///
+    /// let mut vec = Vec::new();
+    /// vec.push(Point::new(40.02f64, 116.34));
+    /// vec.push(Point::new(42.02f64, 116.34));
+    /// vec.push(Point::new(42.02f64, 118.34));
+    /// let linestring = LineString(vec);
+    /// let bbox = linestring.bbox().unwrap();
+    ///
+    /// println!("Bbox top left coordinates: {}, {}", bbox.xmin, bbox.ymax);
+    /// println!("Bbox bottom right coordinates: {}, {}", bbox.xmax, bbox.ymin);
+    /// ```
+    ///
+    fn bbox(&self) -> Option<Bbox<T>>;
+}
+
+fn get_bbox<T>(vect: &Vec<Point<T>>) -> Option<Bbox<T>>
+    where T: Float
```

Comment:
On Mon, Jul 18, 2016 at 6:29 PM, Guillaume P. <notifications@github.com>
wrote:

> In src/algorithm/boundingbox.rs
> <https://github.com/georust/rust-geo/pull/41#discussion_r71180646>:
>
> > +fn get_bbox<T>(vect: &Vec<Point<T>>) -> Option<Bbox<T>>
> > +    where T: Float
>
> Prototype of the function:
>
> fn get_bbox<I, T>(collection: I) -> Option<Bbox<T>>
>     where T: Float,
>           I: IntoIterator<Item = &Point<T>>
>
> Then, the fold version that I've given should almost work. You can also
> write an imperative version using for loop.
>
> You can then rewrite MultiSomething as something that look like:
>
> get_bbox(multi.0.iter().flat_map(|poly| poly.0.iter()))
>
>
I do like the idea to take as input a Point<T> iterator instead of a vector
that it is much more general. Therefore I think this function should be
refactored following your prototype. Actually I've tried to implement it
myself but I'm new to rust and I'm fighting with lifetime and other compile
errors. If someone with more experience on rust want to work/help on this
I'm open to pull requests or hints :-)